crates/uv/src/commands/python/pin.rs

```diff
@@ -56,6 +77,12 @@ pub(crate) async fn pin(
         Err(err) => return Err(err.into()),
     };
 
+    if !isolated {
+        if let Some(ref python) = python {
+            assert_python_compatibility(python).await?;
```

Comment:
Same commentary as https://github.com/astral-sh/uv/pull/4989/files#r1678230289

Unlike when reading the pins, during a write I think we should:

1. Error if the _request_ is incompatible with the Python requirement
2. Warn if the _resolved_ Python is incompatible with the Python requirement (unless `--resolved` is used, then error).

Reply:
Done